internal/buildsign/helper.go

```diff
@@ -0,0 +1,67 @@
+package buildsign
+
+import (
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+)
+
+//go:generate mockgen -source=helper.go -package=buildsign -destination=mock_helper.go
+
+type Helper interface {
+	ApplyBuildArgOverrides(args []kmmv1beta1.BuildArg, overrides ...kmmv1beta1.BuildArg) []kmmv1beta1.BuildArg
+	GetRelevantBuild(moduleBuild *kmmv1beta1.Build, mappingBuild *kmmv1beta1.Build) *kmmv1beta1.Build
+}
+
+type helper struct{}
+
+func NewHelper() Helper {
+	return &helper{}
+}
+
+func (m *helper) ApplyBuildArgOverrides(args []kmmv1beta1.BuildArg, overrides ...kmmv1beta1.BuildArg) []kmmv1beta1.BuildArg {
+	overridesMap := make(map[string]kmmv1beta1.BuildArg, len(overrides))
+
+	for _, o := range overrides {
+		overridesMap[o.Name] = o
+	}
+
+	unusedOverrides := sets.StringKeySet(overridesMap)
+
+	for i := 0; i < len(args); i++ {
+		argName := args[i].Name
+
+		if o, ok := overridesMap[argName]; ok {
+			args[i] = o
+			unusedOverrides.Delete(argName)
+		}
+	}
+
+	for _, overrideName := range unusedOverrides.List() {
+		args = append(args, overridesMap[overrideName])
+	}
+
+	return args
+}
+
+func (m *helper) GetRelevantBuild(moduleBuild *kmmv1beta1.Build, mappingBuild *kmmv1beta1.Build) *kmmv1beta1.Build {
+	if moduleBuild == nil {
+		return mappingBuild.DeepCopy()
+	}
+
+	if mappingBuild == nil {
+		return moduleBuild.DeepCopy()
+	}
+
+	buildConfig := moduleBuild.DeepCopy()
+	if mappingBuild.DockerfileConfigMap != nil {
+		buildConfig.DockerfileConfigMap = mappingBuild.DockerfileConfigMap
+	}
+
+	buildConfig.BuildArgs = m.ApplyBuildArgOverrides(buildConfig.BuildArgs, mappingBuild.BuildArgs...)
+
+	// [TODO] once MGMT-10832 is consolidated, this code must be revisited. We will decide which
```

Comment:
This Jira is an assisted-installer Jira.

Reply:
removed